src/index.js

```diff
@@ -8,12 +8,24 @@ export default function register(Component, tagName, propNames, options) {
 			options && options.shadow ? inst.attachShadow({ mode: 'open' }) : inst;
 		return inst;
 	}
+
 	PreactElement.prototype = Object.create(HTMLElement.prototype);
 	PreactElement.prototype.constructor = PreactElement;
 	PreactElement.prototype.connectedCallback = connectedCallback;
 	PreactElement.prototype.attributeChangedCallback = attributeChangedCallback;
 	PreactElement.prototype.disconnectedCallback = disconnectedCallback;
 
+	// b \ a
+	// take all Component properties and remove existing PreactElement properties
+	const a = new Set(Object.getOwnPropertyNames(PreactElement.prototype));
+	const b = new Set(Object.getOwnPropertyNames(Component.prototype));
```

Comment:
Heh - I'd originally thought of this approach, but then realized `Component.prototype` doesn't includes most of the Preact API methods - it only has `setState` and `forceUpdate`, the rest are defined only by subclasses.